### ISSUE TITLE
Add setting to enable experimental background indexing in SourceKit-LSP

### ISSUE THREAD

**Issue by ahoppen** [swiftlang/vscode-swift#987]
Could you add a setting that allows background indexing to be enabled? Something like

---

**Experimental: Background Indexing**

Enable SourceKit-LSP’s experimental background indexing for SwiftPM projects, providing cross-file functionality without needing to build. This requires SourceKit-LSP bundled with Swift 6.

----

When enabled, send the following in the `initializationOptions` of the `initialize` request.
```json
{
  "backgroundIndexing": true,
  "backgroundPreparationMode": "enabled"
}
```

**Comment by ahoppen** [comment]
Passing `--experimental-feature background-indexing` as the command line argument to sourcekit-lsp has the same behavior as `"backgroundIndexing": true`. We prefer to use the initialization options now because it allows us to remove options. Removing a command line option would cause SourceKit-jLPS to fail to launch.

**Comment by plemarquand** [comment]
Fixed in #992

**Comment by award999** [comment]
Verified with d6c91dd379abd6a402dddb3503e35fd328c6ce21
